src/components/AddItemForm/AddItemForm.js

```diff
@@ -0,0 +1,115 @@
+import { useState } from 'react';
+import { db } from '../../lib/firebase.js';
+
+const AddItemForm = ({ listId }) => {
+  const defaultFormValues = {
+    itemName: '',
+    purchaseInterval: '7',
+  };
+
+  const [formValues, setFormValues] = useState(defaultFormValues);
+
+  // generic function updates formValues state for any of the below form inputs
+  const handleChange = (event) => {
+    const inputName = event.target.name;
+    setFormValues({ ...formValues, [inputName]: event.target.value });
+  };
+
+  // on form submit, add user input and other default values to database
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+
+    const newItem = formValues;
+    newItem.purchaseInterval = parseInt(newItem.purchaseInterval); // convert purchaseInterval to integer
+    newItem.lastPurchaseDate = null;
+    newItem.numberOfPurchases = 0;
+
+    try {
+      await db.collection(`lists/${listId}/items`).add(formValues); // add item to Firestore database
+      setFormValues(defaultFormValues); // after saving to database, reset form values to defaults
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  return (
+    <form name="addItemForm" onSubmit={handleSubmit} className="add-item-form">
+      <label
+        className="add-item-form__label add-item-form__label_type_text label"
+        htmlFor="itemName"
+      >
+        Item name:
+      </label>
+      <input
+        className="add-item-form__text-field text-field"
+        type="text"
+        id="itemName"
+        name="itemName"
+        aria-describedby="itemNameHint"
+        value={formValues.itemName}
+        onChange={handleChange}
+        maxLength="100"
+        required
+      />
+      <span id="itemNameHint"></span>
+
+      <fieldset>
+        <legend>How soon will you buy this again?</legend>
+
+        <input
+          type="radio"
+          id="soonOption"
+          className="add-item-form__radio"
+          name="purchaseInterval"
+          value="7"
+          onChange={handleChange}
+          checked={formValues.purchaseInterval === '7'}
+        />
+        <label
+          htmlFor="soonOption"
+          className="add-item-form__label add-item-form__label_type_radio label"
+        >
+          Soon
+        </label>
+
+        <input
+          type="radio"
+          className="add-item-form__radio"
+          id="kindaSoonOption"
+          name="purchaseInterval"
+          value="14"
+          onChange={handleChange}
+          checked={formValues.purchaseInterval === '14'}
+        />
```

Comment:
For these `purchaseInterval` items, is it possible to simply set them to an actual number? Then we would not have to do this up above: 

`newItem.purchaseInterval = parseInt(newItem.purchaseInterval); // convert purchaseInterval to integer`

Reply:
The reason we used strings is due to input values being string type, so the quickest way we could think to get around that was to just use strings for the purchaseInterval values and comparisons, up until actually putting the value in the database. 

Any suggestions for a better way? I just tried out valueAsNumber, but it seems to return NaN for our radio buttons.